File: app/src/main/java/com/scatl/uestcbbs/module/setting/adapter/OpenSourceAdapter.java
package com.scatl.uestcbbs.module.setting.adapter;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.OpenSourceBean;

/**
 * author: sca_tl
 * description:
 * date: 2020/1/27 13:46
 */
public class OpenSourceAdapter extends BaseQuickAdapter<OpenSourceBean, BaseViewHolder> {

    public OpenSourceAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, OpenSourceBean item) {
        helper.setText(R.id.item_open_source_name, (helper.getAdapterPosition() + 1) + "、" + item.name)
                .setText(R.id.item_open_source_author, "作者：" + item.author)
                .setText(R.id.item_open_source_desc, item.description);
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/setting/adapter/OpenSourceAdapter.kt
package com.scatl.uestcbbs.module.setting.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemOpenSourceBinding
import com.scatl.uestcbbs.entity.OpenSourceBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder

/**
 * Created by sca_tl at 2023/6/9 17:15
 */
class OpenSourceAdapter: PreloadAdapter<OpenSourceBean, ItemOpenSourceBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemOpenSourceBinding {
        return ItemOpenSourceBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemOpenSourceBinding>, position: Int, item: OpenSourceBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.projectName.text = item.name
        holder.binding.authorName.text = item.author
        holder.binding.dsp.text = item.description
    }
}